### DISCUSSION TITLE
当添加一个图例时, 可以保持以前的图例颜色不变吗?

### DISCUSSION THREAD

**Discussion by vaynevayne** [antvis/G2#5008] [Q&A] [answered]
5-11 号的颜色变了

<img width="1100" alt="image" src="https://github.com/antvis/G2/assets/124666577/3a573791-35ef-405a-b8bc-c1c5f11024d5">

<img width="1095" alt="image" src="https://github.com/antvis/G2/assets/124666577/475ab77b-6de2-4f46-8275-28829e640a7c">

有办法做到吗?  比如说固定图例的顺序

**Comment by hustcc** [comment, marked as answer]
可以的，对颜色通道进行回调即可，对于 G2 v4 版本，使用：

```ts
chart
  .line()
  .position('month*temperature')
  .color('city', (city) => city === 'London' ? 'red' : 'blue');  // 👈🏻
```

对于 v5 版本，可以使用：

```ts
chart
  .line()
  .encode('x', (d) => new Date(d.Date))
  .encode('y', 'Close')
  .encode('color', 'Symbol')
  .scale('color', { 
    relations: [['AAPL', '#ff0000']],  // 👈🏻
  })                          
```

**Comment by vaynevayne** [comment]
找到个简单的办法,   事先unshift一条有type,但没值的数据到data里, 这样有图例没数据, 颜色就不会变了 data.unshift({type:'参考值'})
后期 来了参考值 的数据,直接 data.concat ,也不影响

**Reply by hustcc** [reply]
因为默认颜色是按照数据数据顺序 和 色板中颜色值顺序，意义对应，所以改变数据顺序，可以影响颜色~
